streaming/base/converters/dataframe_to_mds.py

```diff
@@ -0,0 +1,326 @@
+            if v not in mds_supported_dtypes:
+                raise ValueError(f'{v} is not supported by MDSwrite')
+        return None
+
+    schema = dataframe.schema
+    schema_dict = {}
+
+    for field in schema:
+        dtype = map_spark_dtype(field.dataType)
+        if dtype in _encodings:
+            schema_dict[field.name] = dtype
+        else:
+            print(_encodings)
+
+    return schema_dict
+
+
+def do_merge_index(partitions: Iterable,
+                   mds_path: Union[str, Tuple[str, str]],
+                   skip: bool = False):
+    """Merge index.json from partitions into one for streaming.
+
+    Args:
+        partitions (Iterable): partitions that contain pd.DataFrame
+        mds_path (Tuple or str): (str,str)=(local,remote), str = local or remote based on parse_uri(url) result
+        skip (bool): whether to merge index from partitions
+    """
+    if not partitions or skip:
+        return
+
+    shards = []
+
+    for row in partitions:
+        mds_partition_index = f'{row.mds_path}/{get_index_basename()}'
+        mds_partition_basename = os.path.basename(row.mds_path)
+        obj = json.load(open(mds_partition_index))
+        for i in range(len(obj['shards'])):
+            shard = obj['shards'][i]
+            for key in ['raw_data', 'zip_data']:
+                if shard.get(key):
+                    basename = shard[key]['basename']
+                    obj['shards'][i][key]['basename'] = os.path.join(mds_partition_basename,
+                                                                     basename)
+        shards += obj['shards']
+
+    obj = {
+        'version': 2,
+        'shards': shards,
+    }
+
+    if isinstance(mds_path, str):
+        mds_index = os.path.join(mds_path, get_index_basename())
+    else:
+        mds_index = os.path.join(mds_path[0], get_index_basename())
+
+    with open(mds_index, 'w') as out:
+        json.dump(obj, out)
+
+
+def dataframeToMDS(dataframe: DataFrame,
+                   merge_index: bool = True,
+                   sample_ratio: float = -1.0,
+                   mds_kwargs: Optional[Dict[str, Any]] = None,
+                   udf_iterable: Optional[Callable] = None,
+                   udf_kwargs: Optional[Dict[str, Any]] = None) -> Tuple[Any, int]:
+    """Execute a spark dataframe to MDS conversion process.
+
+    This method orchestrates the conversion of a spark dataframe into MDS format by
+    processing the input data, applying a user-defined iterable function if
+    provided, and writing the results to MDS-compatible format. The converted data is saved to mds_path.
+
+    Args:
+        dataframe (pyspark.sql.DataFrame or None): A DataFrame containing Delta Lake data.
+        merge_index (bool): Whether to merge MDS index files. Default is True.
+        sample_ratio (float): The fraction of data to randomly sample during conversion.
+            Should be in the range (0, 1). Default is -1.0 (no sampling).
```

Comment:
`Default to` instead of `Default is` since we do this way in our codebase. And wrap the default value in double back tick mark so that the documentation renders correctly. You can refer to any file in streaming code base.